File: Cloud5S_WEB/src/app/@filter/SO/order-scale.filter.ts
import {BaseFilter} from '../Common/base.filter';
import * as moment from 'moment';
export class OrderScaleFilter extends BaseFilter {
  selectedRange: Date[] = [moment().startOf('month').toDate(), moment().endOf('month').toDate()];
  PartnerCode: string = '';
  ItemCode: string = '';
  VehicleCode: string = '';
  code: string = '';
  station: string = '';
  orderType: string = '';
  scaleTypeCode: string[] = [];
  areaCode: string = '';
  companyCode: string = '';
}

export class OrderScaleFilterRequest {
  FromDate?: string | null = moment().startOf('month').format('YYYY-MM-DD');
  ToDate?: string | null = moment().endOf('month').format('YYYY-MM-DD');
  PartnerCode: string | null = '';
  ItemCode: string | null = '';
  VehicleCode: string | null = '';
  companyCode: string | null = '';
  Type?: string[] | null = [];
  currentPage: number | null = 1;
  pageSize: number | null = 20;
  keyWord: string | null = '';
}

export class OrderDetail {
  code?: string = '';
}

File: Cloud5S_API/DMS.Business/Filter/SO/OrderScaleFilter.cs
using DMS.BUSINESS.Filter.Common;

namespace DMS.BUSINESS.Filter.SO
{
    public class OrderScaleFilter : BaseFilter
    {
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public string PartnerCode { get; set; }
        public string ItemCode { get; set; }
        public string VehicleCode { get; set; }
        public string Type { get; set; }
        public string CompanyCode { get; set; }
        public string AreaCode { get; set; }
        public bool Weight1 { get; set; } = false;
        public bool Weight2 { get; set; } = false;

        public bool? IsCanceled { get; set; }
    }
}
